testlib/utils.py

```diff
@@ -76,6 +76,25 @@ def process_exists(name):
     return None
 
 
+def terminate_traces(name):
+    """
+    Terminate trace processes with the given filename.  This is
+    intended for Python scripts whose name will be in the cmdline, but
+    not in the process name.
+    :param name: name of script to clean up
+    :type name: str
+    return: None
+    """
+    procs = []
+    for proc in psutil.process_iter(["cmdline"]):
+        try:
+            if any(name == param for param in proc.info["cmdline"]):
+                for termproc in procs:
```

Comment:
Just get rid of the ```procs``` list entirely. The thing you want to call ```terminate()``` on is just the ```proc``` in ```for proc in psutil.process_iter(...):```.

Reply:
Resolved.